python/mxnet/autograd.py

```diff
@@ -0,0 +1,68 @@
+# coding: utf-8
+"""Autograd for NDArray."""
+from __future__ import absolute_import
+from __future__ import division
+
+import ctypes
+import functools
+from .base import _LIB, check_call
+from .base import mx_uint, NDArrayHandle, c_array
+from .ndarray import NDArray
+from .ndarray import array as ndarray
+
+def set_recording(recording):
+    check_call(_LIB.MXAutogradSetRecording(
+        ctypes.c_int(recording)))
+
+def set_mark_for_record(arrays, mark):
+    handles = []
+    for arr in arrays:
+        handles.append(arr.handle)
+    check_call(_LIB.MXAutogradSetMarkForRecord(
+        len(handles),
+        c_array(NDArrayHandle, handles),
+        ctypes.c_int(mark)))
+
+def compute_gradient(inputs, outputs):
+    input_handles = []
```

Comment:
sorry, you are right, the inputs of ComputeGradient are different with forward inputs